src/main/java/org/opensearch/ad/model/AnomalyDetector.java

```diff
@@ -173,96 +124,68 @@ public AnomalyDetector(
         Instant lastUpdateTime,
         List<String> categoryFields,
         User user,
-        String resultIndex
+        String resultIndex,
+        ImputationOption imputationOption
     ) {
-        if (Strings.isBlank(name)) {
-            throw new ValidationException(CommonMessages.EMPTY_NAME, ValidationIssueType.NAME, ValidationAspect.DETECTOR);
-        }
-        if (Strings.isBlank(timeField)) {
-            throw new ValidationException(CommonMessages.NULL_TIME_FIELD, ValidationIssueType.TIMEFIELD_FIELD, ValidationAspect.DETECTOR);
-        }
-        if (indices == null || indices.isEmpty()) {
-            throw new ValidationException(CommonMessages.EMPTY_INDICES, ValidationIssueType.INDICES, ValidationAspect.DETECTOR);
-        }
+        super(
+            detectorId,
+            version,
+            name,
+            description,
+            timeField,
+            indices,
+            features,
+            filterQuery,
+            windowDelay,
+            shingleSize,
+            uiMetadata,
+            schemaVersion,
+            lastUpdateTime,
+            categoryFields,
+            user,
+            resultIndex,
+            detectionInterval,
+            imputationOption
+        );
+
         if (detectionInterval == null) {
-            throw new ValidationException(
-                ADCommonMessages.NULL_DETECTION_INTERVAL,
-                ValidationIssueType.DETECTION_INTERVAL,
-                ValidationAspect.DETECTOR
-            );
-        }
-        if (invalidShingleSizeRange(shingleSize)) {
-            throw new ValidationException(
-                "Shingle size must be a positive integer no larger than " + TimeSeriesSettings.MAX_SHINGLE_SIZE + ". Got " + shingleSize,
-                ValidationIssueType.SHINGLE_SIZE_FIELD,
-                ValidationAspect.DETECTOR
-            );
+            errorMessage = ADCommonMessages.NULL_DETECTION_INTERVAL;
+            issueType = ValidationIssueType.DETECTION_INTERVAL;
```

Comment:
the issueType could've been set to ValidationIssueType.NAME in Config class when super is called, then it returns and lets say detector interval is also null, it will reset the issueType to ValidationIssueType.DETECTION_INTERVAL. While the sequential order matters less here cause they will all need to be fixed eventually we can have a check that issueType is null before continuing with the rest of the checks or just be okay with it being over-written.

Reply:
I changed to

```
public AnomalyDetector(
        String detectorId,
        Long version,
        String name,
        String description,
        String timeField,
        List<String> indices,
        List<Feature> features,
        QueryBuilder filterQuery,
        TimeConfiguration detectionInterval,
        TimeConfiguration windowDelay,
        Integer shingleSize,
        Map<String, Object> uiMetadata,
        Integer schemaVersion,
        Instant lastUpdateTime,
        List<String> categoryFields,
        User user,
        String resultIndex,
        ImputationOption imputationOption
    ) {
        super(
            detectorId,
            version,
            name,
            description,
            timeField,
            indices,
            features,
            filterQuery,
            windowDelay,
            shingleSize,
            uiMetadata,
            schemaVersion,
            lastUpdateTime,
            categoryFields,
            user,
            resultIndex,
            detectionInterval,
            imputationOption
        );

        checkAndThrowValidationErrors();

        if (detectionInterval == null) {
            errorMessage = ADCommonMessages.NULL_DETECTION_INTERVAL;
            issueType = ValidationIssueType.DETECTION_INTERVAL;
        } else if (((IntervalTimeConfiguration) detectionInterval).getInterval() <= 0) {
            errorMessage = ADCommonMessages.INVALID_DETECTION_INTERVAL;
            issueType = ValidationIssueType.DETECTION_INTERVAL;
        }

        int maxCategoryFields = ADNumericSetting.maxCategoricalFields();
        if (categoryFields != null && categoryFields.size() > maxCategoryFields) {
            errorMessage = CommonMessages.getTooManyCategoricalFieldErr(maxCategoryFields);
            issueType = ValidationIssueType.CATEGORY;
        }

        checkAndThrowValidationErrors();

        this.detectorType = isHC(categoryFields) ? MULTI_ENTITY.name() : SINGLE_ENTITY.name();
    }

    private void checkAndThrowValidationErrors() {
        if (errorMessage != null && issueType != null) {
            throw new ValidationException(errorMessage, issueType, ValidationAspect.DETECTOR);
        } else if (errorMessage != null || issueType != null) {
            throw new TimeSeriesException(CommonMessages.FAIL_TO_VALIDATE);
        }
    }

```

Reply:
same can be done for forecaster

Reply:
yes, done it for forecaster